packages/autocomplete/index.ts

```diff
@@ -60,6 +61,26 @@ async function completer(params: AutocompleteParameters, line: string): Promise<
   const CONFIG_COMPLETIONS = shellSignatures.ShellConfig.attributes as TypeSignatureAttributes;
   const SHARD_COMPLETE = shellSignatures.Shard.attributes as TypeSignatureAttributes;
 
+  const splitLineWhitespace = line.split(' ');
+  const command = splitLineWhitespace[0];
+  if (SHELL_COMPLETIONS[command]?.isDirectShellCommand) {
+    // If we encounter a direct shell commmand, we know that we want completions
+    // specific to that command, so we set the 'exclusive' flag on the result.
```

Comment:
Why is this necessary? Will this also work in the browser?

Reply:
@mcasimir Are you referring to the `'exclusive'` flag here?

The mongosh-repl.ts changes have a somewhat more specific context here, but the gist of it is that we run both the Node.js default autocompleter and our own there and combine the results; but the Node.js REPL thinks that it should complete e.g. `use a` to `use assert`, because it just looks at the `a` and finds global variables with the same name.

> Will this also work in the browser? 

I don’t think the extra array entry should hurt, we’re only using the first entry anyway:

https://github.com/mongodb-js/mongosh/blob/831220a316738c0d28f4ce464c83ba56b8b8c75d/packages/browser-runtime-core/src/autocompleter/shell-api-autocompleter.ts#L22